xmltest.cpp

```diff
@@ -431,7 +435,8 @@ int main( int argc, const char ** argv )
 		XMLTest( "Programmatic DOM", 2, doc->FirstChildElement()->LastChildElement( "sub" )->IntAttribute( "attrib" ) );
 		XMLTest( "Programmatic DOM", "& Text!",
 				 doc->FirstChildElement()->LastChildElement( "sub" )->FirstChild()->ToText()->Value() );
-		XMLTest("User data", (void*)2 == comment->GetUserData(), true, false);
+		XMLTest("User data", true, &dummyValue == comment->GetUserData(), false);
+		XMLTest("User data", dummyInitialValue, dummyValue, false);

```

Comment:
I like the fix, but this doesn't really test anything. &dummyValue == GetUserData() (previous line) makes sense. But for this line, no data has been read back. It's testing that the assignment worked. Is there a missing variable assignment?